pkg/xray/resource_xray_settings.go

```diff
@@ -0,0 +1,99 @@
+package xray
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceXraySettings() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceXrayDbSyncTimeCreate,
+		ReadContext:   resourceXrayDbSyncTimeRead,
+		UpdateContext: resourceXrayDbSyncTimeUpdate,
+		DeleteContext: resourceXrayDbSyncTimeDelete,
+		Description:   "Provides an Xray DB Sync Time resource.",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_sync_updates_time": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The time of the Xray DB sync daily update job.",
+				ValidateDiagFunc: matchesHoursMinutesTime,
+			},
+		},
+	}
+}
+
+type DbSyncDailyUpdatesTime struct {
+	DbSyncTime string `json:"db_sync_updates_time"`
+}
+
+func unpackDBSyncTime(s *schema.ResourceData) DbSyncDailyUpdatesTime {
+	d := &ResourceData{s}
+	dbSyncTime := DbSyncDailyUpdatesTime{
+		DbSyncTime: d.getString("db_sync_updates_time", false),
+	}
+	return dbSyncTime
+}
+
+func packDBSyncTime(dbSyncTime DbSyncDailyUpdatesTime, d *schema.ResourceData) diag.Diagnostics {
+	if err := d.Set("db_sync_updates_time", dbSyncTime.DbSyncTime); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func getDbSyncTime(client *resty.Client) (DbSyncDailyUpdatesTime, *resty.Response, error) {
+	dbSyncTime := DbSyncDailyUpdatesTime{}
+	resp, err := client.R().SetResult(&dbSyncTime).Get("xray/api/v1/configuration/dbsync/time")
+	return dbSyncTime, resp, err
+}
+
+// No create functionality provided by API. DB sync time exists by default and can only be modified
+func resourceXrayDbSyncTimeCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	return resourceXrayDbSyncTimeUpdate(ctx, d, m)
```

Comment:
Don't do this - literally in the schema config use the same function for both. You don't need this function at all

Reply:
ok, removed. My thought was it will be more clear what and why we are doing it